send-transaction-service/src/send_transaction_service.rs

```diff
@@ -597,10 +597,12 @@ impl SendTransactionService {
                     .last_sent_time
                     .map(|last| now.duration_since(last) >= retry_rate)
                     .unwrap_or(false);
-                if !nonce_account::verify_nonce_account(&nonce_account, &durable_nonce)
-                    && signature_status.is_none()
-                    && expired
-                {
```

Comment:
This logic ensure the nonce transaction to have allocated time to finish before being dropped altogether. Without this check, the transaction might be dropped as expired whenever the retry loop is run even right after the transaction was sent in the send thread.

Reply:
Mm, to confirm, you were seeing situations where the nonce had been advanced from a successful transaction but the signature_status wasn't available yet?

Reply:
I am still not sure why we keep onto the transaction even if `verify_nonce_account` returns false?

Reply:
`verify_nonce_account` returns false when the nonce has been advanced, which happens on both successful and failed durable-nonce transactions.
I think @lijunwangs is saying he saw instances where a successful tx would return None for the check on L592 because the signature wasn't yet available, but where the nonce had already been advanced by the time `verify_nonce_account` ran on L600, which means the tx should have been preserved to be checked until root. But best for him to confirm that I'm understanding correctly.